### ISSUE TITLE
Cleanup tests

### ISSUE THREAD

**Issue by vkostyukov** [finagle/finch#156]
We should be comprehensive in the test style we use. I'm suggesting the following style:
* `class XSpec extends FlatSuite with Matchers` - a configuration
* `x shouldBe y` - an assertion
* `a [x] should be thrownBy y` - excepted exceptions

I should probably create a document `CONTRIBUTING.md` that addresses these items and other ideas.

**Comment by rpless** [comment]
How do you feel about testing Lists or Seqs? Currently we use the following:
```scala
result should have length 1
result should contain("25")
```
Do we want to keep doing this, or do we want to use something like:
```scala
result shouldBe List("25")
```

**Comment by vkostyukov** [comment]
Hm. Good question. The first variant looks fluent and works great for the big sequences. But we usually deal with small seqs/lists. So, I think the second variant is shorted and fits well into the whole style. What do you think?

**Comment by rpless** [comment]
We don't currently have any large Seqs in tests and I think the second form fits better with our style. I'll do it that way when I make the PR, and if we decided we don't want it, I can change it.

**Comment by rpless** [comment]
Also we could be do:
```scala
    a [ParamNotFound] shouldBe thrownBy(Await.result(futureResult))
```
Instead of 
```scala
    a [ParamNotFound] should be thrownBy Await.result(futureResult)
```
I kinda like the first better, but the second probably unifies the style better. Thoughts?

**Comment by vkostyukov** [comment]
Can we left the parents out in the first example?

**Comment by rpless** [comment]
It seems like the answer is no. The compiler expects a semicolon but founds a period at `Await.result`.

**Comment by vkostyukov** [comment]
The first one fits the style better. But the second one is already used everywhere. Do you think it's a pain to replace it with the first variant? I would go with the first variant if it's possible.

**Comment by rpless** [comment]
I can do the first variant. I'll also update contributing.md to reflect this.

**Comment by vkostyukov** [comment]
Cool! Thanks a lot Ryan!

**Comment by vkostyukov** [comment]
Fixed by #175.
